### PR TITLE
Add search component with autocomplete to navigation bar

### PR DESCRIPTION
**Pull request by colbyfayock**
This adds a search form / input to the navigation bar.

I set up a fuse instance that filters searches on query, giving a new fuzzy search experience. This can easily be updated to include more keys as we apply more details to the search index, such as categories.

I'm decoding the HTML entities of the search index for the title, as it's provided in the rendered form. We're not using it as such in that instance as directly, we're plugging it into a JSON object.

I ended up adding the navbar to the homepage for at least a temporary option to alleviate not having search there. I'm not super bothered by it there, but i liked the clean look without it

Resolves https://github.com/colbyfayock/next-wordpress-starter/issues/1

### CODE REVIEW

File: src/components/Nav/Nav.js [hunk 1]
```diff
@@ -1,14 +1,103 @@
-import styles from './Nav.module.scss';
+import { useEffect, useRef } from 'react';
+import Link from 'next/link';
 
 import useSite from 'hooks/use-site';
+import useSearch from 'hooks/use-search';
```
Comment (doingandlearning) [inline src/components/Nav/Nav.js:5]: I love how you're using the hook here.

File: src/components/Nav/Nav.js [hunk 2]
```diff
@@ -1,14 +1,103 @@
...
+
+    return () => removeDocumentOnClick();
+  }, [query]);
+
+  /**
+   * addDocumentOnClick
+   */
+
+  function addDocumentOnClick() {
+    document.body.addEventListener('click', handleOnDocumentClick, true);
```
Comment (doingandlearning) [inline src/components/Nav/Nav.js:40]: I've never seen this before - I'm stealing it 😀

File: src/hooks/use-search.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react';
+import Fuse from 'fuse.js';
```
Comment (doingandlearning) [inline src/hooks/use-search.js:2]: I love how clear this hook is, very grokable!

File: src/pages/index.js [hunk 4]
```diff
@@ -10,25 +10,12 @@ import Section from 'components/Section';
...
-import styles from 'styles/Home.module.scss';
+import styles from 'styles/pages/Home.module.scss';
 
 export default function Home({ posts }) {
   const { metadata = {} } = useSite();
   const { name, description } = metadata;
 
-  console.log('searchIndex', searchIndex);
-
-  useEffect(() => {
```
Comment (doingandlearning) [inline src/pages/index.js:23]: Nothing like deleted code.

**Review by doingandlearning — APPROVED** [review]
Sorry, nothing to add here. Top notch!

**Comment by colbyfayock** [comment]
thanks for all the feedback @doingandlearning 🙌

**Pull request merged**